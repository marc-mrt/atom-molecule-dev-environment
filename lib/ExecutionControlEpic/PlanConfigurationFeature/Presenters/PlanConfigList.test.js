import "raf/polyfill";
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";
import { PlanConfigList } from "./PlanConfigList";

Enzyme.configure({ adapter: new Adapter() });

describe("PlanConfigList", () => {
  let spy = jest.fn();
  let subject = shallow(
    <PlanConfigList
      values={[
        { name: "NODE_PLAN", value: "web" },
        { name: "NODE_ENV", value: "dev" },
      ]}
      items={{
        type: "object",
        schemas: {
          name: {
            type: "string",
            default: "",
            placeholder: "name",
          },
          value: {
            type: "string",
            default: "",
            placeholder: "value",
          },
        },
      }}
      onChange={spy}
    />,
  );
  it("should display the values", () => {
    expect(subject).toMatchSnapshot();
  });
});
