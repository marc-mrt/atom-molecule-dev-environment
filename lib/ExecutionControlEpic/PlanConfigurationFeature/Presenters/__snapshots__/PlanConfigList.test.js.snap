// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanConfigList should display the values 1`] = `
<styled.div>
  <styled.span
    error={false}
  />
  <FormWatcher
    __debug={false}
    watchPath={[Function]}
  />
</styled.div>
`;
